src/ansys/mechanical/core/embedding/app.py

```diff
@@ -245,9 +241,20 @@ def plot(self) -> None:
             warnings.warn("Plotting is only supported with version 2024R2 and later!")
             return
 
-        from ansys.mechanical.core.embedding.viz.pyvista_plotter import plot_model
+        # TODO Check if anything loaded inside app or else show warning and return
+
+        from ansys.mechanical.core.embedding.viz.embedding_plotter import to_plotter
 
-        plot_model(self)
+        return to_plotter(self)
+
+    def plot(self) -> None:
+        """Visualize the model in 3d.
+
+        Requires installation using the viz option. E.g.
+        pip install ansys-mechanical-core[viz]
+        """
+        _plotter = self.plotter()
+        return _plotter.show()
```

Comment:
Since we are making a requirement in the docstring, it would be nice to add a warning and an early return here too, in case the user call this function without having the required dependencies.

```suggestion
    def plot(self) -> None:
        """Visualize the model in 3d.

        Requires installation using the viz option. E.g.
        pip install ansys-mechanical-core[viz]
        """
        if not HAS_ANSYS_VIZ:
            warnings.warn(
                "Installation of viz option required! Use pip install ansys-mechanical-core[viz]"
            )
            return
        
        _plotter = self.plotter()
        return _plotter.show()
```

Reply:
Since plotter() already checks this  and it get calls inside the plot() , I feel its not necessary